python3.13-slack-sdk-3.35.0: Causes build failure.

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

_Edit: looks like it was ntfy causing the issue._
I did check an there were no other completed or open requests on python3.13-slack-sdk-3.35.0. To be honest I don't have it in my config and I'm not sure what is calling for it. But is is causing `sudo nixos-rebuild switch` to fail. 

I think this might be the last python issue since things got changed around. 

For reference here are the python pkgs I have in no order:

```
    python312Packages.python-lsp-server
    python312Packages.opentype-feature-freezer
    python312Packages.fonttools
    kdePackages.kdev-python
    python312Packages.websockets
    python314
    python312Packages.ws4py
    python312Packages.wsproto
    python313Packages.pywal
    python313Packages.python-lsp-ruff
    python312Packages.docopt
    python312Packages.numpy
    python312Packages.pyaudio
```


### Steps to reproduce

Run sudo nixos-rebuild switch

### Expected behaviour

Build succsessfully 

### Screenshots

_No response_

### Relevant log output

```console
building the system configuration...
error: builder for '/nix/store/wxlmqx982b61xypq8wq26m65id72zw3w-python3.13-slack-sdk-3.35.0.drv' failed with exitcode 1;
last 25 log lines:
> tests/slack_sdk_async/socket_mode/test_interactions_aiohttp.py::TestInteractionsAiohttp::test_interactions
> tests/slack_sdk_async/socket_mode/test_interactions_aiohttp.py::TestInteractionsAiohttp::test_interactions_with_disconnection
> tests/slack_sdk_async/socket_mode/test_interactions_aiohttp.py::TestInteractionsAiohttp::test_interactions_with_disconnection
>   /build/source/slack_sdk/socket_mode/aiohttp/__init__.py:377: DeprecationWarning: ssl=None is deprecated, please use ssl=True
>     self.current_session = await self.aiohttp_client_session.ws_connect(
>
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_coverage
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_legacy_coverage
>   /build/source/slack_sdk/web/internal_utils.py:293: UserWarning: The top-level `text` argument is missing in the request payload for a chat.postEphemeral call - It's a best practice to always provide a `text` argument when posting a message. The `text` argument is used in places where content cannot be rendered such as: system push notifications, assistive technology such as screen readers, etc.
>     warnings.warn(missing_text_message, UserWarning)
>
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_coverage
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_legacy_coverage
>   /build/source/slack_sdk/web/internal_utils.py:293: UserWarning: The top-level `text` argument is missing in the request payload for a chat.postMessage call - It's a best practice to always provide a `text` argument when posting a message. The `text` argument is used in places where content cannot be rendered such as: system push notifications, assistive technology such as screen readers, etc.
>     warnings.warn(missing_text_message, UserWarning)
>
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_coverage
> tests/slack_sdk_async/web/test_web_client_coverage.py::TestWebClientCoverage::test_legacy_coverage
>   /build/source/slack_sdk/web/internal_utils.py:293: UserWarning: The top-level `text` argument is missing in the request payload for a chat.update call - It's a best practice to always provide a `text` argument when posting a message. The `text` argument is used in places where content cannot be rendered such as: system push notifications, assistive technology such as screen readers, etc.
>     warnings.warn(missing_text_message, UserWarning)
>
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> =========================== short test summary info ============================
> FAILED tests/slack_sdk_async/socket_mode/test_aiohttp.py::TestAiohttp::test_init_with_loop - RuntimeError: Session and connector has to use same event loop
> = 1 failed, 849 passed, 5 skipped, 7 deselected, 307 warnings in 237.30s (0:03:57) =
For full logs, run:
nix log /nix/store/wxlmqx982b61xypq8wq26m65id72zw3w-python3.13-slack-sdk-3.35.0.drv
error: 1 dependencies of derivation '/nix/store/z1r7ix4rn8h761ppdp8lyqv8pvky7vqf-ntfy-2.7.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cni1pcahyq95hk46hxg473gp7q9s5xx7-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pq0bqfvn8hab78ahvlh3l0jfgkixjnpz-ntfy-2.7.0_fish-completions.drv'failed to build
error: 1 dependencies of derivation '/nix/store/qmi26r427jk3wng33wjj5qajlqmrj0wl-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xcs0mjnwag43v8c3fl51dp9z98ndicgg-nixos-system-nixos-25.11.20250621.4206c4c.drv' failed to build
Command '['nix', '--extra-experimental-features', 'nix-command flakes', 'build', '--print-out-paths', '/etc/nixos#nixosConfigurations."nixos".config.system.build.toplevel', '--no-link']' returned non-zero exit status 1.
```

### Additional context

As I said I checked before posing this bug report. If its being handled and I missed it I apologize.  And I'm sorry if its not much to go on.

Here is the full log from `nix log /nix/store/wxlmqx982b61xypq8wq26m65id72zw3w-python3.13-slack-sdk-3.35.0.drv` It was to long to fit in the box above.

https://bin.garudalinux.org/?22667bde5d813c97#HYxJGKpGyMBkexuBtk21ngK3gKp8xsG82RNS7GBTFzS4

### System metadata

- system: `"x86_64-linux"`
- host os: `Linux 6.16.0-rc1-cachyos, NixOS, 25.11 (Xantusia), 25.11.20250613.ee930f9`
- multi-user?: `yes`
- sandbox: `yes`
- version: `nix-env (Nix) 2.28.3`
- channels(root): `""`
- nixpkgs: `/nix/store/xnr2xjs0clkhlhqfc3vqcmhnnwdk3c5p-source`

### Notify maintainers

@fabaff

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/slack-sdk/default.nix
+++ b/pkgs/development/python-modules/slack-sdk/default.nix
@@ -7,9 +7,7 @@
   buildPythonPackage,
   fetchFromGitHub,
   fetchpatch,
-  flake8,
   moto,
-  psutil,
   pytest-asyncio,
   pytestCheckHook,
   setuptools,
@@ -41,10 +39,9 @@ buildPythonPackage rec {
 
   build-system = [ setuptools ];
 
-  dependencies = [
+  optional-dependencies.optional = [
     aiodns
     aiohttp
-    aiosqlite
     boto3
     sqlalchemy
     websocket-client
@@ -54,12 +51,11 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "slack_sdk" ];
 
   nativeCheckInputs = [
-    flake8
+    aiosqlite
     moto
-    psutil
     pytest-asyncio
     pytestCheckHook
-  ];
+  ] ++ optional-dependencies.optional;
 
   disabledTests = [
     # Requires internet access (to slack API)

--- a/pkgs/development/python-modules/slack-sdk/default.nix
+++ b/pkgs/development/python-modules/slack-sdk/default.nix
@@ -6,6 +6,7 @@
   boto3,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
   flake8,
   moto,
   psutil,
@@ -29,6 +30,14 @@ buildPythonPackage rec {
     tag = "v${version}";
     hash = "sha256-yjYpALyHSTLQSuwd6xth7nqfi3m1C9tqnWrrVRmI220=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-aiohttp-test_init_with_loop.patch";
+      url = "https://github.com/slackapi/python-slack-sdk/pull/1697.patch";
+      hash = "sha256-rHaJBH/Yxm3Sz/jmzc4G1pVJJXz0PL2880bz5n7w3ck=";
+    })
+  ];
 
   build-system = [ setuptools ];
 

